java/piranha/src/main/java/com/uber/piranha/XPFlagCleaner.java

```diff
@@ -139,12 +149,14 @@
    */
   private int endPos = DONTCARE;
 
-  /* information provided in the config file */
-  private final HashSet<String> treatedMethods = new HashSet<String>();
-  private final HashSet<String> controlMethods = new HashSet<String>();
-  private final HashSet<String> deleteMethods = new HashSet<String>();
-  private final HashSet<String> treatmentGroupMethods = new HashSet<String>();
-  private final HashSet<String> handledAnnotations = new HashSet<String>();
+  /** Information provided in the properties.json config file */
+  private final Set<Map<String, Object>> methodProperties = new HashSet<>();
+
+  private final Map<String, List<Map<String, Object>>> treatedMethods = new HashMap<>();
```

Comment:
Can you reduce the number of maps here to one? The earlier code also had these maps unnecessarily.

Reply:
👍 it will make the data structure one level deeper but makes sense.